src/SourceLocator/Type/DirectorySourceLocator.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace BetterReflection\SourceLocator\Type;
+
+use BetterReflection\Identifier\Identifier;
+use BetterReflection\Identifier\IdentifierType;
+use BetterReflection\Reflector\Reflector;
+use BetterReflection\SourceLocator\Exception\InvalidDirectory;
+
+/**
+ * This source locator loads all php files in an entire directory or multiple directories.
+ */
+class DirectorySourceLocator implements SourceLocator
+{
+    /**
+     * @var AggregateSourceLocator
+     */
+    private $aggregatedSourceLocator;
+
+    /**
+     * @param $directories string[] directories to scan
+     */
+    public function __construct(array $directories)
+    {
+        $sourceLocators = [];
+        foreach ($directories as $dir) {
```

Comment:
There's no need to shorten variable names here, we prefer to use full variable names, e.g. `$directory`, `$recursiveDirectoryIterator` (instead of `$rdi`)